lib/moab/storage_object_validator.rb

```diff
@@ -0,0 +1,196 @@
+# There is repeating code in here - first pass for audit check - storageobjectvalidator
+
+require 'moab'
+
+module Moab
+  # Given a druid path, are the contents actually a well-formed Moab?
+  class StorageObjectValidator
+
+    EXPECTED_VERSION_SUB_DIRS = ["data", "manifests"].freeze
+    EXPECTED_DATA_SUB_DIRS = ["content", "metadata"].freeze
+    IMPLICIT_DIRS = ['.', '..'].freeze # unlike Find.find, Dir.entries returns these
+
+    # error codes
+    INCORRECT_DIR = 0
+    MISSING_DIR = 1
+    EXTRA_CHILD_DETECTED = 2
+    EMPTY = 3
+    NO_SIGNATURE_CATALOG = 4
+    NO_MANIFEST_INVENTORY = 5
+    NO_XML_FILES = 6
+    VERSIONS_NOT_IN_ORDER = 7
+    FILES_IN_VERSION_DIR = 8
+
+    RESPONSE_CODE_TO_MESSAGES = {
+      INCORRECT_DIR=> "Incorrect items in path",
+      MISSING_DIR => "Missing directory: %{addl}",
+      EXTRA_CHILD_DETECTED => "Unexpected item in path: %{addl}",
+      EMPTY => "No items in path",
+      FILES_IN_VERSION_DIR => "Should contain only sequential version directories. Also contains files: %{addl}",
+      NO_SIGNATURE_CATALOG => "Version: %{addl} Missing signatureCatalog.xml",
+      NO_MANIFEST_INVENTORY => "Version: %{addl} Missing manifestInventory.xml",
+      NO_XML_FILES => "Version: %{addl} Missing all required metadata files",
+      VERSIONS_NOT_IN_ORDER => "Should contain only sequential version directories. Current directories: %{addl}"
+    }.freeze
+
+    attr_reader :storage_obj_path
+
+    def initialize(storage_object)
+      @storage_obj_path = storage_object.object_pathname
+      @directory_entries_hash = {}
+    end
+
+    def validate_object
+      results = []
```

Comment:
perhaps 'errors' is more appropriate than 'results' if any message added indicates error.  then the code will explain itself better and many of the comments become unnecessary?